### ISSUE TITLE
I'm using async api, but when cluster node timeout, it will synchronously connect cluster node when updating route

### ISSUE THREAD

**Issue by lcygosky** [vipshop/hiredis-vip#122]

**Comment by thakurajayL** [comment]
Thats right and it gets blocked forever. Chances are there that your heart-bit monitoring task might kill the process if its stuck.
I had done code to solve this issue but did not commit and unfortunately I lost the code as well.
1. When route table is required due to any error call the async callback in the application.
2. This async callback should initiate raw command  (0 second timer) to get "cluster nodes"  from the Redis cluster.
   Now Cluster nodes command can be fired to any one of the nodes (cluster mode)
3. When response is received Application can feed the response to redis lib and it can then update its routing table.
